content/docs/projects/discounts.md

```diff
@@ -0,0 +1,270 @@
+
+# 2 Maintenance
+
+## 2.1 Child DocTypes
+
+Discounts are maintained in the master data associated to them as a child table. They are generally separated into two child tables: Buying Discounts and Selling Discounts, depending on which ones are applicable to the DocType.
+
+The following fields are typically stored in these child DocTypes:
+
+- For Price List (`for_price_list`): Link to Price List
+- Supplier (`supplier`): Link to Supplier (hidden for Supplier discounts)
+- Supplier Name (`supplier_name`): Supplier Name fetched from the `supplier` linked (hidden for Supplier discounts)
+- Discount Type (`discount_type`): Select field (Compounding, On Total)
+- Discount Percentage (`discount_percentage`): Float
+- Valid From (`valid_from`): Date
+- Valid Up To (`valid_upto`): Date
+
+Currently, the following are the child DocTypes for discounts:
+
+- Item Buying Discounts
+- Item Selling Discounts
+- Supplier Buying Discounts
+- Customer Selling Discounts
+- Principal Buying Discounts
+
+## 2.2 Validations
+
+Validations should be put in place to make sure that only valid values are entered in the fields.
+
+### 2.2.1 Item Discounts
+
+For Item Discounts, the validation of Buying and Selling Discounts are separated into two functions/methods:
+
+```python
+from erpnext.selling.utils import validate_selling_discounts
+
+class Item(WebsiteGenerator):
+    def validate(self):
+        self.validate_item_buying_discounts()
+        validate_selling_discounts(self)
+
+    def validate_item_buying_discounts(self):
+        ...
+        # Validate that percentages are between 0 and 100
+        # Validate that valid_upto > valid_from
+```
+
+## 2.2.2 Supplier Discounts
+
+For Supplier Discounts, only Buying Discounts are available:
+
+```python
+class Supplier(TransactionBase):
+    def validate(self):
+        self.validate_supplier_buying_discounts()
+
+    def validate_supplier_buying_discounts(self):
+        ...
+        # Validate that percentages are between 0 and 100
+        # Validate that valid_upto > valid_from
+```
+
+## 2.2.3 Customer Discounts
+
+For Customer Discounts, only Selling Discounts are available:
+
+```python
+from erpnext.selling.utils import validate_selling_discounts
+
+class Customer(TransactionBase):
+    def validate(self):
+        validate_selling_discounts(self)
+```
+
+## 2.2.4 Principal Discounts
+
+For Principal Discounts, only Buying Discounts are available:
+
+```python
+class Principal(Document):
+    def validate(self):
+        self.validate_discounts()
+
+    def validate_discounts(self):
+        for item in self.buying_discounts:
+            item.validate_discount_values()
+
+class PrincipalBuyingDiscount(Document):
+    def validate_discount_values(self):
+        # If discount is Actual on Total, checks if discount amount is not negative
+        # If discount is On Total, Compounding, checks if discount percentage is not negative
+```
```

Comment:
I think it will be better  to put on what level each discount applies to? 
example:  Discounts maintained in item profile means that the discount is applied on an item level. 

Something like that hehe

Reply:
Done (f4c4dbd420dc2e4097a7f0102fc42c10aa76f42b)

![image](https://github.com/BizKit-Tech/BizKit-Tech.github.io/assets/62210335/8d651580-ed4b-4b4b-bf92-58f7a5451e2a)